File: PythonSM/BlinkSM.py
from Enums import BlinkStates
from SM import SM
import RPi.GPIO as GPIO

class BlinkSM(SM):
	def __init__(self, ledPin, elapsedTime, period):
		super().__init__(elapsedTime, period)
		self.ledPin = ledPin
		self.State = BlinkStates.OFF

	def Step(self):
		#transition
		if self.State == BlinkStates.OFF:
			self.State = BlinkStates.ON
		else: 
			self.State = BlinkStates.OFF	
		#action
		if self.State == BlinkStates.OFF:
			GPIO.output(self.ledPin, GPIO.LOW)
		else: 	
			GPIO.output(self.ledPin, GPIO.HIGH)

File: C++SM/src/blinkSM.cpp
#include <wiringPi.h>
#include <iostream>
#include "../headers/blinkSM.h"

using namespace std;

BlinkSM::BlinkSM(string name, unsigned long elapsedTime, unsigned long period, int ledPin): SM(name, elapsedTime, period)
{
    this->ledPin = ledPin;
    this->state = B_INIT;
    digitalWrite (this->ledPin, 0); 
}

void BlinkSM::Step()
{
    // transition
    switch(this->state)
    {
        case B_INIT:
            this->state = B_OFF;
            break;
        case B_OFF:
            this->state = B_ON;
            break;
        case B_ON:
            this->state = B_OFF;
            break;
        default:
            break;
    }

    // action
    switch(this->state)
    {
        case B_OFF:
            digitalWrite (this->ledPin, 0); 
            break;
        case B_ON:
            digitalWrite (this->ledPin, 1); 
            break;
        default:
            break;
    }
}

void BlinkSM::Stop()
{
    digitalWrite (this->ledPin, 0); 
}